core/tee/tee_svc_cryp.c

```diff
@@ -3409,47 +3465,83 @@ static TEE_Result get_sm2_kep_params(const TEE_Attribute *params,
 			found |= BIT(IS_INITIATOR);
 			break;
 		case TEE_ATTR_ECC_PUBLIC_VALUE_X:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
-					     p->content.ref.length,
+			res = memdup_user(p->content.ref.buffer,
+					  p->content.ref.length, &bufs[n]);
+			if (res)
+				return res;
+
+			crypto_bignum_bin2bn(bufs[n], p->content.ref.length,
```

Comment:
We should free `bufs[n]` as soon as we're done with it to minimize heap utilization.

Same below for the other bignums.